woocommerce/woocommerce
# Clear only plugin cache if WC header are not present to load with WC headers.

### All Submissions:






### Changes proposed in this Pull Request:

WC headers are added in filter `extra_plugin_headers`, however, in case when WooCommerce is activated/updated, `get_plugins` will be called and cache will be set before this filter can be cached.

Earlier we were also clearing the update cache, which was causing issue with showing update notices (see #26348), this only clears the plugin cache.

Also, `get_plugins` call is expensive even with an update cache present, because it does many I/O calls, so we should clear the cache very conservatively.



Closes #26348.

### How to test the changes in this Pull Request:

1. Install an outdated plugin ([example](https://downloads.wordpress.org/plugin/akismet.4.1.zip)) (you can download the previous version of a plugin from WordPress.org > Plugin page > Advanced view ).
2. Update to WooCommerce 4.1 RC2 from any older WooCommerce version
3. Make sure that update prompt for your outdated plugin is still displayed.

### Other information:




### Changelog entry

> Fix - Clear only plugin cache after check if headers are not present.
